src/lib/operators/join_hash.cpp

```diff
@@ -221,22 +221,25 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
       PerformanceWarning("Build relation larger than probe relation in hash join");
     }
 
-    const auto l2_cache_size = 256'000;  // bytes
+    const auto l2_cache_size = 1'024'000;  // bytes 
+    const auto l2_cache_max_usable = l2_cache_size * 0.5;  // use 50% of the L2 cache size  
 
-    // To get a pessimistic estimation (ensure that the hash table fits within the cache), we assume
-    // that each value maps to a PosList with a single RowID. For the used small_vector's, we assume a
-    // size of 2*RowID per PosList. For sizing, see comments:
+    // For sizing, see comments:
     // https://probablydance.com/2018/05/28/a-new-fast-hash-table-in-response-to-googles-new-fast-hash-table/
+    // We aim for a fill factor of 80%, a bit less of bytell hash map's maximum fill factor of 0.9375.
```

Comment:
Why 80%? Does the post in the URL explain that?

Reply:
It's large enough to avoid unnecessary oversizing but still be small enough to cause unexpected resizings. That's all.
This is rather a very simple heuristic. I wouldn't know how to measure a better value. :-/